pkg/crosscluster/logical/sql_row_reader_test.go

```diff
@@ -140,3 +141,54 @@ func TestSQLRowReader(t *testing.T) {
 		readRowsSql(t, dbDest, primaryKeys),
 		"reading destination did not yield expected rows")
 }
+
+func TestSQLRowReaderWithArrayColumn(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+
+	srv, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	defer srv.Stopper().Stop(ctx)
+	s := srv.ApplicationLayer()
+	runner := sqlutils.MakeSQLRunner(sqlDB)
+
+	// Create tables with array column
+	createStmt := `CREATE TABLE tab_array (pk int[] primary key, value int)`
+	runner.Exec(t, createStmt)
+
+	// Insert test data
+	runner.Exec(t, "INSERT INTO tab_array VALUES (ARRAY['1', '2'], 10), (ARRAY['1'], 20)")
+
+	// Create sqlRowReader for source table
+	desc := desctestutils.TestingGetPublicTableDescriptor(s.DB(), s.Codec(), "defaultdb", "tab_array")
+	session := newInternalSession(t, s)
+	defer session.Close(ctx)
+	reader, err := newSQLRowReader(ctx, desc, session)
+	require.NoError(t, err)
+
+	db := s.InternalDB().(isql.DB)
+
+	testRows := []tree.Datums{
+		{
+			tree.NewDArrayFromDatums(types.Int, tree.Datums{tree.NewDInt(1), tree.NewDInt(2)}),
+			tree.NewDInt(0),
```

Comment:
these `tree.NDInt(0)` guys are just dummy non-pk values, correct? it's not strictly necessary to pass non-pk column values into `reader.ReadRows()`, right? No need to change anything, just want to confirm my understanding.

Reply:
Yep! These values are there to simulate stale values in a read refresh. They could be DNULL or any other int value.